src/core/imap/MCIMAPSession.cpp

```diff
@@ -832,6 +832,15 @@ void IMAPSession::login(ErrorCode * pError)
     else if (r == MAILIMAP_ERROR_PARSE) {
         mShouldDisconnect = true;
         * pError = ErrorParse;
+
+        String * response = MCSTR("");
+        if (mImap->imap_response != NULL) {
+            response = String::stringWithUTF8Characters(mImap->imap_response);
+        } else if (mImap->imap_stream_buffer != NULL) {
+            response = String::stringWithUTF8Characters(mImap->imap_stream_buffer->str);
```

Comment:
I think we should create a separate accessor for that buffer and keep is as raw Data * since we're not guaranteed that it will convert smoothly as UTF8.